### PR TITLE
Fix for phpunit tests

### PR DESCRIPTION
**Pull request by cklingsporn**
Adjusted application class used for testing 
Formevents::PRE_BIND deprec and so changed to PRE_SUBMIT
Also context change in basketvalidator

phpunit passing

### CODE REVIEW

File: tests/ProductBundle/Command/GenerateProductCommandTest.php [hunk 1]
```diff
@@ -68,9 +69,10 @@ public function testExecute()
      */
     private function getCommandInstance()
     {
-        $kernel = $this->getMockBuilder('Symfony\Component\HttpKernel\Kernel')->disableOriginalConstructor()->getMock();
+        $kernel = $this->getMockBuilder(Kernel::class)->disableOriginalConstructor()->getMock();
```
Comment (Soullivaneuh) [inline tests/ProductBundle/Command/GenerateProductCommandTest.php]: Don't use `::class`

File: tests/ProductBundle/Command/GenerateProductCommandTest.php [hunk 2]
```diff
@@ -68,9 +69,10 @@ public function testExecute()
...
         $app = new Application($kernel);
         $app->add(new GenerateProductCommand());
+
```
Comment (Soullivaneuh) [inline tests/ProductBundle/Command/GenerateProductCommandTest.php]: No needed new blank line.

File: tests/Component/Form/BasketValidatorTest.php [hunk 3]
```diff
@@ -32,7 +32,20 @@ public function testValidate()
 
         $consValFact = $this->getMockBuilder('Symfony\Bundle\FrameworkBundle\Validator\ConstraintValidatorFactory')->disableOriginalConstructor()->getMock();
 
-        $context = $this->getMockBuilder('Symfony\Component\Validator\Context\ExecutionContext')->disableOriginalConstructor()->getMock();
+        /**
```
Comment (Soullivaneuh) [inline tests/Component/Form/BasketValidatorTest.php]: This should not be a phpdoc AFAIK. cc @greg0ire

File: tests/Component/Form/BasketValidatorTest.php [hunk 4]
```diff
@@ -32,7 +32,20 @@ public function testValidate()
...
+         * Symfony 2.4+ moves ExecutionContext into sub-namespace of .\context
+         * So if the class exists then we should use this one else it's probably 2.3 
+         *
```
Comment (Soullivaneuh) [inline tests/Component/Form/BasketValidatorTest.php]: Blank line to be removed.

File: tests/Component/Form/BasketValidatorTest.php [hunk 5]
```diff
@@ -32,7 +32,20 @@ public function testValidate()
...
         $consValFact = $this->getMockBuilder('Symfony\Bundle\FrameworkBundle\Validator\ConstraintValidatorFactory')->disableOriginalConstructor()->getMock();
 
-        $context = $this->getMockBuilder('Symfony\Component\Validator\ExecutionContext')->disableOriginalConstructor()->getMock();
+        /**
+         * Symfony 2.4+ moves ExecutionContext into sub-namespace of .\context
+         * So if the class exists then we should use this one else it's probably 2.3 
+         * 
+         */
+        if (class_exists('Symfony\Component\Validator\Context\ExecutionContext')) {
+        	$context = $this->getMockBuilder('Symfony\Component\Validator\Context\ExecutionContext')->disableOriginalConstructor()->getMock();
```
Comment (Soullivaneuh) [inline tests/Component/Form/BasketValidatorTest.php]: Please respect CS. No tabs.

File: tests/Component/Form/BasketValidatorTest.php [hunk 6]
```diff
@@ -32,7 +32,20 @@ public function testValidate()
...
+
+        } else {
+        	$context = $this->getMockBuilder('Symfony\Component\Validator\ExecutionContext')->disableOriginalConstructor()->getMock();
```
Comment (Soullivaneuh) [inline tests/Component/Form/BasketValidatorTest.php]: No tabs.

File: tests/Component/Form/BasketValidatorTest.php [hunk 7]
```diff
@@ -32,7 +32,20 @@ public function testValidate()
...
+        }
+        
+        
+
```
Comment (Soullivaneuh) [inline tests/Component/Form/BasketValidatorTest.php]: Two useless lines.

File: .travis.yml [hunk 8]
```diff
@@ -47,7 +47,7 @@ before_script:
   - composer config -q github-oauth.github.com $GITHUB_OAUTH_TOKEN
   - if [ "$SYMFONY_VERSION" = "2.8.*@dev" ] || [ "$SYMFONY_VERSION" = "3.0.x-dev as 2.8" ]; then SYMFONY_DEPRECATIONS_HELPER=strict; fi;
   - if [ "$SYMFONY_VERSION" != "" ]; then composer require "symfony/symfony:${SYMFONY_VERSION}" --no-update; fi;
-  - travis_wait composer update --prefer-dist --no-interaction $COMPOSER_FLAGS
+  - travis_wait 40 composer update --prefer-dist --no-interaction $COMPOSER_FLAGS
```
Comment (Soullivaneuh) [inline .travis.yml]: Why?

**Comment by Soullivaneuh** [comment]
Lot of trailing white spaces on your code. Could you please fix them?

File: tests/Component/Form/BasketValidatorTest.php (hunk 3 above)
Comment (greg0ire) [inline tests/Component/Form/BasketValidatorTest.php]: Nope, it shouldn't, please remove one star

File: tests/Component/Form/BasketValidatorTest.php [hunk 9]
```diff
@@ -32,7 +32,20 @@ public function testValidate()
 
         $consValFact = $this->getMockBuilder('Symfony\Bundle\FrameworkBundle\Validator\ConstraintValidatorFactory')->disableOriginalConstructor()->getMock();
 
-        $context = $this->getMockBuilder('Symfony\Component\Validator\Context\ExecutionContext')->disableOriginalConstructor()->getMock();
+        /**
+         * Symfony 2.4+ moves ExecutionContext into sub-namespace of .\context
```
Comment (greg0ire) [inline tests/Component/Form/BasketValidatorTest.php]: What is this `.\` about ?

File: tests/Component/Form/BasketValidatorTest.php (hunk 3 above)
Comment (greg0ire) [inline tests/Component/Form/BasketValidatorTest.php]: Shouldn't `phpdoc_to_comment` have been triggered?

**Comment by cklingsporn** [comment]
Apologies, didn't realise PR auto updates on branch.

File: .travis.yml (hunk 8 above)
Comment (cklingsporn) [inline .travis.yml]: Looks like initial composer update takes a large amount of time on travis.
The initial update for php 5.5, php 5.6 took around 40 minutes, after the inital cache they are running at 4-8 minutes.
Currently trying to work out another way of getting php 5.3 & 5.4 up and running (they time out at the 2 hour travis cap)

File: tests/Component/Form/BasketValidatorTest.php (hunk 3 above)
Comment (Soullivaneuh) [inline tests/Component/Form/BasketValidatorTest.php]: ecommerce is not unde StyleCI yet.

File: .travis.yml (hunk 8 above)
Comment (Soullivaneuh) [inline .travis.yml]: Ok let's keep it for now. Will be normalized by dev-kit later.

**Comment by cklingsporn** [comment]
Seems that travis_wait 40 minutes not enough for Travis initial build.

Probably reliant on load somewhere, but looks like php 5.4 takes ~41 minutes and 5.3 takes 40-80 minutes.

Both do pass once the initial composer update is done.

File: composer.json [hunk 10]
```diff
@@ -18,13 +18,14 @@
         "symfony/console": "~2.3",
         "symfony/process": "~2.3",
         "symfony/twig-bridge": "~2.3",
-
+        "symfony/intl": "^2.6|~3",
```
Comment (Soullivaneuh) [inline composer.json]: This project support Symfony 2.3 and seems to not support 3.0 for now.

**Comment by Soullivaneuh** [comment]
Sorry for that, but the actual master will be replaced by develop.

See: https://github.com/sonata-project/dev-kit/issues/138#issuecomment-228303637

This PR will be auto-closed.

But if the issue is still here on the new master, we can re-consider it after that. :wink:

**Pull request closed without merging**